packages/router/src/Set.tsx

```diff
@@ -46,7 +46,7 @@ export function Set<WrapperProps>(props: SetProps<WrapperProps>) {
     ...rest
   } = props
   const routerState = useRouterState()
-  const location = useLocation()
```

Comment:
I did a little research into this when I was writing up a ticket. When running a test to see if it did what was expected.

The location provider would get caught in an infinite loop due to this code 
```javascript
  componentDidMount() {
    this.HISTORY_LISTENER_ID = gHistory.listen(() => {
      this.setState(() => ({ context: this.getContext() }))
    })
  }
  ```
  So removing  `useLocation` should prevent that.
  
  @Tobbe I think one thing that this PR doesn't cover is using just `<Private />` I no we now recommned using `<Set private />` but we didn't deprecate `Private` so I think that this fix only fixes it for the `Set` component

Reply:
This is the implementation of `<Private>`

```jsx
export function Private<WrapperProps>(props: PrivateProps<WrapperProps>) {
  const { children, unauthenticated, role, wrap, ...rest } = props

  return (
    <Set<any>
      private
      unauthenticated={unauthenticated}
      role={role}
      wrap={wrap}
      {...rest}
    >
      {children}
    </Set>
  )
}
```

As you can see it uses `<Set>` under the hood, so the fix covers `<Private>` too

Reply:
Ahhh okay I didn't know it did that under the hood. Is it worth adding a test case for the `Private` component just in case this changes in the future. Not sure it ever would But who knows what the future holds